projects/core/src/configurator/commons/facade/configurator-groups.service.ts

```diff
@@ -22,20 +22,82 @@ export class ConfiguratorGroupsService {
     private configuratorGroupStatusService: ConfiguratorGroupStatusService
   ) {}
 
+  /**
+   * Returns the current group Id.
+   * In case no group Id is being set before returns the first group of the configuration.
+   * Return null when configuration contains no groups.
+   *
+   * @param owner configuration owner
+   */
   getCurrentGroupId(owner: GenericConfigurator.Owner): Observable<string> {
     return this.configuratorCommonsService.getConfiguration(owner).pipe(
       map((configuration) => {
         if (configuration?.interactionState?.currentGroup) {
           return configuration.interactionState.currentGroup;
         } else {
-          return configuration?.groups?.length > 0
-            ? configuration.groups[0].id
+          return configuration?.flatGroups?.length > 0
+            ? configuration.flatGroups[0].id
             : null;
         }
       })
     );
   }
 
+  /**
+   * Return the first conflict group of a configuration.
+   *
+   * @param configuration - Configuration
+   */
+  getFirstConflictGroup(
+    configuration: Configurator.Configuration
+  ): Configurator.Group {
+    return configuration.flatGroups.find(
+      (group) => group.groupType === Configurator.GroupType.CONFLICT_GROUP
+    );
+  }
+
+  /**
+   * Navigates to the first non-conflict group of the configuration which is not completed.
+   * This method assumes that the configuration has uncompleted groups,
+   * the caller has to verify this prior to calling this method.
+   *
+   * @param owner - Configuration owner
+   */
+  goToFirstUncompletedGroup(owner: GenericConfigurator.Owner): void {
+    this.configuratorCommonsService
```

Comment:
Is it not better to defibe, for instance, **configuration$** Observable taht is equal to **this.configuratorCommonsService.getConfiguration(owner)** and simple access previously defined **configuration$** Observable that to call **this.configuratorCommonsService.getConfiguration(owner)**. Direct access to **configuration$** Observable is more readable.

Reply:
Do I understand it correctly, that I should add a step, 
let configuration$ = this.configuratorCommonsService.getConfiguration(owner);
configuration$..pipe(take(1))
      .subscribe(( .......
?
I disagree. To introduce a new variable in this method would only make the code longer but not more readable.

Reply:
Resolved